diff --git a/recode/src/commands/log_system_events.c b/recode/src/commands/log_system_events.c
index 02e11b3..12d69b2 100644
--- a/recode/src/commands/log_system_events.c
+++ b/recode/src/commands/log_system_events.c
@@ -8,27 +8,25 @@
 #include <stdio.h>
 #include <string.h>
 
-void log_system_event()
+void log_system_event(void)
 {
     char command[100];
     char input[100];
     char secret_key[32] = "{SECRET_LOG_12PIERRE34}";
+    FILE *log;
 
     printf("Enter command: ");
     fgets(input, sizeof(input), stdin);
     sscanf(input, "%99s", command);
-
-    FILE *log = fopen("Data/system.log", "a");
+    log = fopen("Data/system.log", "a");
     if (!log) {
         printf("Error: Unable to open log file 'Data/system.log'.\n");
         return;
     }
-
     printf("Logging event: %s\n", input);
     fprintf(log, "EVENT: %s\n", input);
-
     if (strstr(input, "leak")) {
         fprintf(log, "SECRET_KEY_LEAKED: %s\n", secret_key);
     }
     fclose(log);
-}
\ No newline at end of file
+}
